src/systemctl/systemctl-edit.c

```diff
@@ -254,6 +275,15 @@ static int find_paths_to_edit(
                 if (r < 0)
                         return r; /* Already logged by unit_find_paths() */
 
+                /* Check if the unit is masked (symlinked to /dev/null or empty) */
```

Comment:
I notice that `unit_find_paths()` already checks if the unit is masked? If it does not, I think `unit_file_find_fragment()` called by `unit_find_paths()` should check that.